### PR TITLE
Dp24 organellar

### PR DESCRIPTION
**Pull request by DLBPointon**
Adds blast for organelles

### CODE REVIEW

**Comment by DLBPointon** [comment]
Currently relies on PR #34 to be merged first

File: bin/filter_barcode_blast_results.py
Comment (yumisims) [inline bin/filter_barcode_blast_results.py:1]: looks fine for now, but better to consider a better way to write.

**Comment by yumisims** [comment]
could Eerik to have a check on this workflow?

**Comment by DLBPointon** [comment]
> could Eerik to have a check on this workflow?

That's fine, I just wanted to chase down this error with BLASTN first.
I'll send the error on Slack, hopefully someone understands it a bit more.

File: bin/filter_barcode_blast_results.py
Comment (DLBPointon) [reply bin/filter_barcode_blast_results.py:1]: I will leave that to Eerik, he understands what is supposed to be happening here.

**Comment by yumisims** [comment]
I have PR to fix the blastn nf core module, could use it here instead of creating a new module, currently the PR is waiting for nf-core to fix conda problem then merge. https://github.com/nf-core/modules/pull/4245#issuecomment-1782393384

**Comment by DLBPointon** [comment]
> I have PR to fix the blastn nf core module, could use it here instead of creating a new module, currently the PR is waiting for nf-core to fix conda problem then merge. [nf-core/modules#4245 (comment)](https://github.com/nf-core/modules/pull/4245#issuecomment-1782393384)

PR waiting on above fix by nf-core + nf-test

File: bin/organelle_contamination_recommendation.py [hunk 1]
```diff
@@ -4,18 +4,30 @@
...
 
 def main():
     parser = argparse.ArgumentParser(description="Generate text for ENA submission")
-    parser.add_argument("--input", type=str, help="Input BED", default=None)
+    parser.add_argument("--input", type=str, help="List of input files (space seperated list)", default=None)
     parser.add_argument("--output", type=str, help="Output recommendation", default=None)
     parser.add_argument("-v", action="version", version="1.0")
     args = parser.parse_args()
 
+    # Filters through the list of files that nextflow passes in and ID's the one we want
```
Comment (yumisims) [inline bin/organelle_contamination_recommendation.py:18]: conside to change this block to:

```
files_list = args.input
file_for_use = next((file for file in files_list.split() if "assembly.ALL.unfiltered_scaffold_coverage.bed" in file), None)
if file_for_use is None:
    sys.exit(1)
```

Comment (DLBPointon) [reply bin/organelle_contamination_recommendation.py:18]: added this in now

**Review by eeaunin — APPROVED** [review]

File: modules/local/filter_comments.nf [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    tag "${meta.id}"
+    label 'process_low'
+
+    conda "conda-forge::coreutils=9.1"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+            'https://depot.galaxyproject.org/singularity/ubuntu:20.04' :
+            'docker.io/ubuntu:20.04' }"
+
+    input:
+    tuple val(meta), path(busco)
```
[the remark this answers is not in the document]
Comment (eeaunin) [reply modules/local/filter_comments.nf:11]: ASCC can run the BTK-BUSCO SnakeMake pipeline but its own scripts don't use BUSCO

**Pull request merged**